lib/src/git.rs

```diff
@@ -637,15 +637,27 @@ impl<'a> RemoteCallbacks<'a> {
                 return Ok(creds);
             } else if let Some(username) = username_from_url {
                 if allowed_types.contains(git2::CredentialType::SSH_KEY) {
-                    if std::env::var("SSH_AUTH_SOCK").is_ok()
-                        || std::env::var("SSH_AGENT_PID").is_ok()
-                    {
-                        tracing::debug!(username, "using ssh_key_from_agent");
-                        return git2::Cred::ssh_key_from_agent(username).map_err(|err| {
-                            tracing::error!(err = %err);
-                            err
-                        });
+                    // Try to get the SSH key from the agent by default, and report an error
+                    // only if it _seems_ like that's what the user wanted.
+                    //
+                    // Note that the env variables read below are **not** the only way to
+                    // communicate with the agent, which is why we request a key from it no
+                    // matter what.
+                    match git2::Cred::ssh_key_from_agent(username) {
+                        Ok(key) => return Ok(key),
```

Comment:
Would it make sense to keep the `tracing::debug` message in case of success?

Reply:
Done.